soc: egis_et171: update config and resolve issue of ram func

jackylee-go opened the pull request:
There are four modificaions of following commits.
1. Remove the default configuration "[FLASH_LOAD_SIZE](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/egis/egis_et171/Kconfig.defconfig#L6)" from "[Kconfig.defconfig](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/egis/egis_et171/Kconfig.defconfig)".
Because of it is only used by "linker.ld" and it will automatically assign the [same default value](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/arch/riscv/common/linker.ld#L46-L53) as "Kconfig.defconfig".

2. Add a "__weak" declaration to "[soc_early_init_hook](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/egis/et171/soc.c#L10)" function.
This is because "soc_early_init_hook" is already implemented in the et171 SOC.
By overriding the weak function, applications or external modules can implement their own "soc_early_init_hook".

3. To fixed the i-cache activation order issue 
Because ANDES's family of SOCs enables cache in [reset.S](https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/riscv/custom/andes/reset.S#L24-L39), this execution precedes z_data_copy() (which copies the flash code to RAM).
That might cause the i-cache to fetch instructions from SRAM before the d-cache writes them back to SRAM.
To prevent this, executing "fence.i" instruction to force cache synchronization.

4. Updated boards/index.rst
Modify index.rst according to the suggestions in [#91940](https://github.com/zephyrproject-rtos/zephyr/pull/91940#discussion_r2409553150)

Commit:
boards: egis_et171: Remove unnecessary default configuration

Remove the default configuration FLASH_LOAD_SIZE, because
"linker.ld" will automatically configure FLASH_LOAD_SIZE
to the same default value

Signed-off-by: Jacky Lee <jacky.lee@egistec.com>

jackylee-go commented:
@niedzwiecki-dawid @nordicjm @jimmyzhe @kartben
It seems the CI didn't assign any reviewer.
Could you kindly take a moment to review it? Much appreciated!

maass-hamburg commented on soc/egis/et171/soc.c:
```diff
@@ -7,8 +7,16 @@
 {
+#if CONFIG_XIP && CONFIG_ICACHE
+	/* Since caching is enabled before z_data_copy(), RAM functions may still be cached
+	 * in the d-cache instead of being written to SRAM. In this case, the i-cache will
```
Could you create a general issue for that?
Then I'm fine with the quick fix.

jimmyzhe replied:
Thanks @jackylee-go for the feedback.
This issue also affects AE350, so I created #98046 to record it.

I am ok with this fix too.

jackylee-go commented on soc/egis/et171/soc.c:
```diff
@@ -7,8 +7,16 @@
  #include <zephyr/kernel.h>
 
-void soc_early_init_hook(void)
+__weak void soc_early_init_hook(void)
```
Currently, other drivers that haven't issue a PR have been completed and verified as modules.
However, some implementations require to modify 'soc_early_init_hook', which will cause conflicts when using upstream to verify.
Add a 'weak' declaration can avoid those conflicts.

jimmyzhe replied:
The clock requirement should be solved by implement a clock driver or add Egis Kconfig option to set the clock.

nordicjm replied:
@andyross @peter-mitsis for comment on making `soc_early_init_hook` weak

jackylee-go replied:
This is why I emphasize that not all drivers for this SOC are fully implemented.
This declaration is added solely to facilitate smoother development of subsequent drivers.
If the relevant driver has already been developed, this declaration can be removed.

If this approach is not appropriate for upstream , please let me know.
I’ll try to find an alternative solution.

maass-hamburg replied:
use SYS_INIT with INIT_LEVEL_EARLY, thats even happening before soc_early_init_hook or use board_early_init_hook , that is directly after soc_early_init_hook

In general if you are developing drivers, that you plan to contribute work in-tree

Commit:
soc: egis_et171: Fix the i-cache activation order issue

Since caching is enabled before z_data_copy(), RAM
functions may still be cached in the d-cache instead
of being written to SRAM. In this case, the i-cache
will fetch the wrong content from SRAM. Thus, using
"fence.i" to fix it.

Signed-off-by: Jacky Lee <jacky.lee@egistec.com>

Commit:
doc: boards: update index.rst

Updated boards/index.rst to remove unnecessary context
and fix statements.

Signed-off-by: Jacky Lee <jacky.lee@egistec.com>
